src/main/java/com/commercetools/sync/commons/utils/CompletableFutureUtils.java

```diff
@@ -0,0 +1,162 @@
+        @Nonnull final List<T> values,
+        @Nonnull final Function<T, CompletionStage<S>> mapper) {
+        return mapValuesToFutureOfCompletedValues(values.stream(), mapper);
+    }
+
+    /**
+     * Creates a Future containing a list of value results after the mapper function is applied to each one and
+     * completed.
+     *
+     * @param values stream of values to apply a mapper function that would map each to a {@link CompletionStage}.
+     * @param mapper function to map each value to a {@link CompletionStage}
+     * @param <T>    The type of the values.
+     * @param <S>    The type of the mapping of the values.
+     * @return a future containing a list of completed stage results of the values after the mapper function was
+     *         applied to each one.
+     */
+    @Nonnull
+    public static <T, S> CompletableFuture<Stream<S>> mapValuesSetToFutureOfCompletedValues(
+        @Nonnull final Set<T> values,
+        @Nonnull final Function<T, CompletionStage<S>> mapper) {
+        return setOfFuturesToFutureOfSet(mapValueSetToFutureSet(values, mapper))
+            .thenApply(Set::stream);
+    }
+
+    /**
+     * Creates a Future containing a list of value results after the mapper function is applied to each one and
+     * completed.
+     *
+     * @param values stream of values to apply a mapper function that would map each to a {@link CompletionStage}.
+     * @param mapper function to map each value to a {@link CompletionStage}
+     * @param <T>    The type of the values.
+     * @param <S>    The type of the mapping of the values.
+     * @return a future containing a list of completed stage results of the values after the mapper function was
+     *         applied to each one.
+     */
+    @Nonnull
+    public static <T, S> CompletableFuture<List<S>> mapValuesToFutureOfCompletedValues(
+        @Nonnull final Stream<T> values,
+        @Nonnull final Function<T, CompletionStage<S>> mapper) {
+        return listOfFuturesToFutureOfList(mapValuesToFutureList(values, mapper));
+    }
+
+    /**
+     * Transforms a set of {@code CompletionStage} into a {@code CompletionStage} of a set,
+     * that will be completed once all the elements of the given set are completed.
+     * In case multiple stages end exceptionally only one error is kept.
+     *
+     * @param set set of {@code CompletionStage}
+     * @param <T> the element obtained from the set of {@code CompletionStage}
+     * @return the {@code CompletableFuture} of a set of elements
+     */
+    @Nonnull
+    public static <T> CompletableFuture<Set<T>> setOfFuturesToFutureOfSet(
+        @Nonnull final Set<? extends CompletionStage<T>> set) {
+
+        final Set<CompletableFuture<T>> futureSet = set.stream()
+                                                       .map(CompletionStage::toCompletableFuture)
+                                                       .collect(toSet());
```

Comment:
minor optimization:
it doesn't have sense right here to collect to slower and more sophisticated `Set`, a `List` could be efficient enough.

And then later collect to `Set` in the return result